File: product-client-java-kotlin/src/main/java/de/byteleaf/microservice/product/control/ProductService.java
package de.byteleaf.microservice.product.control;

import de.byteleaf.microservice.product.client.ProductClient;
import de.byteleaf.microservice.product.entity.Product;
import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class ProductService {

    @Autowired
    private ProductClient productClient;

    @Autowired
    private ModelMapper modelMapper;

    public List<Product> products() {
        return productClient.allProducts().stream()
                .map(it -> modelMapper.map(it, Product.class)).collect(Collectors.toList());
    }

    public Product product(String id) {
        return modelMapper.map(productClient.getProduct(id), Product.class);
    }
}

File: product-client-kotlin/src/main/kotlin/de/byteleaf/microservice/product/control/ProductService.kt
package de.byteleaf.microservice.product.control

import de.byteleaf.microservice.product.client.ProductClient
import de.byteleaf.microservice.product.entity.Product
import org.modelmapper.ModelMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ProductService {

    @Autowired
    private lateinit var productClient: ProductClient

    @Autowired
    private lateinit var modelMapper: ModelMapper

    fun products(): List<Product> {
        return productClient.allProducts().map { modelMapper.map(it, Product::class.java) }
    }

    fun product(id: String): Product {
        return modelMapper.map(productClient.getProduct(id), Product::class.java)
    }
}
